### DISCUSSION TITLE
pyright: eglot-rename putting text in the wrong place ocassionally

### DISCUSSION THREAD

**Discussion by ElleNajt** [joaotavora/eglot#1373] [Q&A]
I'm running into a similar bug in python with pyright lsp to this one: https://github.com/joaotavora/eglot/issues/4

The gist is that sometimes eglot-rename will stick the variable name in extra places, or below the line where it's supposed to happen, or a few characters offset.

It's been really tricky to reproduce, since it happens rarely and the circumstances under which it happens seem really brittle. Would love to submit an MRE, but stuck on how to do this efficiently.

I'm using doom emacs with:
```
(python
        +lsp
        +pyright
        +tree-sitter)  
and 
(lsp +eglot )
```
in my init.el

Thanks!

**Comment by joaotavora** [comment]
> which is easier to reproduce. 

OK, if this one is easier to reproduce, then please do the following:

* change the title of this discussion so that it either somehow reflects the two issues or only that one (if you chose the latter, you can always make another discussion about the hard-to-repro issue later on).
* follow *ALL* the steps in https://joaotavora.github.io/eglot/#Reporting-bugs-1, preferably (but not necessarily) with a recent build of Emacs master.  if you use some older version of Emacs + the Eglot package make sure you specify exactly which versions.

Attention to detail in crafting a reproducible recipe matters a lot here.  You _will_ need to setup a non-Doom emacs. It's not as hard as it may sound though.  usually having the LSP server program (e.g. pyright) and some GNU Elpa packages (say, Eglot and Company)  installed, then running:

`path/to/emacs -Q --package-initialize path/to/some/python/file.py -f eglot`

will be enough for you to show the problem, collect the needed files, and make a beautiful MRE.

The outcome is often one of the following:

* while setting up the Emacs -Q run that demoes the problem you notice something in your local configuration (which from my perspective include anything which is not Emacs proper, i.e. any part of Doom) that was indirectly/subtly causing the problem.  You can nevertheless describe it here and I may be able to help you (or maybe I won't).
* you succeed in showing the problem with an Emacs -Q run and then I can reproduce it here, too, and I can usually fix the problem for you.

**Reply by ElleNajt** [reply]
I was convinced that the behavior in the GIF is actually desirable and not a bug, since the workflow of auto import is that you're writing a call to the function and want it to automatically get put in the imports. So I will remove it from the question. Sorry for including two questions in one.

On the other hand, the thing I'm describing with eglot-rename is definitely a bug, I will try to see if I can reproduce it when I get some time.
